.github/workflows/leapfrog-py-test.yaml

```diff
@@ -0,0 +1,87 @@
+name: Test
+
+on:
+  pull_request:
+    branches:
+      - main
+      - python-interface-poc
+  push:
+    branches:
+      - main
+      - python-interface-poc
+
+jobs:
+  run:
+
+    runs-on: ${{ matrix.config.os }}
+
+    name: ${{matrix.config.os }} (${{ matrix.config.py }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: macos-latest,   py: '3.12'}
+          - {os: ubuntu-latest,  py: '3.8'}
+          - {os: ubuntu-latest,  py: '3.10'}
+          - {os: ubuntu-latest,  py: '3.11'}
+          - {os: ubuntu-latest,  py: '3.12'}
+          - {os: windows-latest, py: '3.12'}
+
+    defaults:
+      run:
+        working-directory: leapfrog-py
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.config.py }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.config.py }}
+
+      - name: Install Eigen3 Linux
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt-get update
+          sudo apt-get install libeigen3-dev
+
+      - name: Install Eigen3 macOS
+        if: runner.os == 'macOS'
+        run: |
+          brew update
+          brew install eigen
+
+      - name: Setup Eigen3 Windows
+        if: runner.os == 'Windows'
+        run: |
+          choco install eigen
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install hatch
+
+      - name: Test
+        run: |
+          hatch run cov-ci
+
+      - name: Lint
+        run: |
+          # https://github.com/python/mypy/issues/10863
+          mkdir -p .mypy_cache
+          hatch run lint:all
```

Comment:
happy with this but might be nicer to run lint before tests, feel like its more immediate feedback

Reply:
Yeah makes sense, tests are actually faster than lint atm but will keep in mind as that probably won't always be true

Reply:
damn tests being faster than lint is amazing